src/components/List/List.tsx

```diff
@@ -1,90 +1,157 @@
+  // @ts-ignore
+  const state = useListState<SelectOption>({ ...props, selectionMode: 'single' });
+  const myRef = React.useRef(null);
+  const combinedRefs = useCombinedRefs(myRef, ref);
+  /** ignore the change on the `onSelectionChange` from any to unknown **/
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const { listBoxProps } = useListBox(props, state, combinedRefs);
+
+  const firstKey = state.collection.getFirstKey();
+  const first = firstKey ? state.collection.getItem(firstKey) : null;
+
+  return (
+    <div css={wrapperStyle({ width })}>
+      <div data-testid={dataTestId ? `${dataTestId}_list` : 'ictinus_list'}>
+        <div style={{ height: height || '100%' }}>
+          <Window
+            {...listBoxProps}
+            css={listStyle({ width, height })}
+            id={listBoxProps.id}
+            isVirtualizationEnabled={isVirtualized}
+            rowHeight={first?.props.rowSize === 'compact' ? 40 : 56}
+            ref={combinedRefs}
+          >
+            {Array.from(state.collection).map((item) => {
+              return item.type === 'section' ? (
+                <ListBoxSection key={item.key} section={item} state={state} />
+              ) : (
+                <Option key={item.key} item={item} state={state} />
+              );
+            })}
+          </Window>
+        </div>
       </div>
-    );
-  }
-);
+    </div>
+  );
+});
 List.displayName = 'List';
 
-export default memo(List, isEqual);
+function Option({ item, state, style }: { item: any; state: any; style?: any }) {
+  // Get props for the option element
+  const ref = React.useRef(null);
+  const { optionProps, isDisabled } = useOption({ key: item.key }, state, ref);
+
+  // Determine whether we should show a keyboard
+  // focus ring for accessibility
+  const { isFocusVisible, isFocused, focusProps } = useFocusRing();
+
+  return (
+    <ListItemWrapper
+      {...mergeProps(optionProps, focusProps)}
+      id={optionProps.id}
+      ref={ref}
+      isDisabled={isDisabled}
+      style={style}
+      data-focus-visible={isFocusVisible || undefined}
+      data-focused={isFocused || undefined}
+      data-empty={state.collection.size === 0 || undefined}
+      rowSize={item.props?.rowSize || 'normal'}
+    >
+      {item.rendered}
+    </ListItemWrapper>
+  );
+}
+
+function ListBoxSection({ section, state }: any) {
+  const { itemProps, headingProps, groupProps } = useListBoxSection({
+    heading: section.rendered,
+    'aria-label': section['aria-label'],
+  });
+
+  return (
+    <>
+      <ListItemWrapperStyled
+        {...itemProps}
+        rowSize={section.props?.rowSize || 'normal'}
+        isDisabled={false}
+      >
+        {section.rendered && (
+          <span {...headingProps} id={headingProps.id}>
+            {section.rendered}
+          </span>
+        )}
+        <ul
+          {...groupProps}
+          aria-labelledby={groupProps['aria-labelledby']}
+          style={{
```

Comment:
maybe move this to `.styles.ts`

Reply:
solved at [`3b16bc4` (#655)](https://github.com/Orfium/orfium-ictinus/pull/655/commits/c11cdfc3afa1dda0bb001b051c48883e1a5b3675)